src/app/views/preferences/preferences.component.ts

```diff
@@ -7,9 +8,12 @@ import { Component, OnInit } from '@angular/core';
 })
 export class PreferencesComponent implements OnInit {
 
-  constructor() { }
+  iconS: IconService;
+
+  constructor(private iconService: IconService) { }
 
   ngOnInit(): void {
+    this.iconS = this.iconService;
```

Comment:
same as above, please try adding the service as public member when using the constructor